pages/verify-eth/[discordServerId]/[discordMemberId]/[customLink].tsx

```diff
@@ -0,0 +1,211 @@
+};
+
+const getSignatureErrorMessage = (
+  error: string
+): {
+  short: string;
+  advanced?: string;
+} => {
+  if (error.includes("Contract not found") || error.includes("UNINITIALIZED"))
+    return {
+      short:
+        "Your wallet is not yet initialized, please make a transaction (sending ETH to yourself works) to initialize it",
+      advanced: error,
+    };
+  return {
+    short: "Your signature could not be verified, please try again",
+    advanced: error,
+  };
+};
+
+const VerifyEthPage = ({
+  discordServerName,
+  discordServerIcon,
+  starknetNetwork,
+}: Props) => {
+  const router = useRouter();
+  const { discordServerId, discordMemberId, customLink } = router.query;
+  const {
+    account,
+    connectWallet,
+    disconnect,
+    networkType,
+    balance,
+    signMessage,
+  } = useWalletConnection();
+
+  const {
+    verifyingSignature,
+    verifiedSignature,
+    unverifiedSignature,
+    verifySignature,
+  } = useSignatureVerification(
+    starknetNetwork,
+    discordServerId as string,
+    discordMemberId as string,
+    customLink as string
+  );
+
+  const sign = useCallback(async () => {
+    if (!account) return;
+    try {
+      const messageCopy = {
+        ...messageToSign,
+        domain: {
+          ...messageToSign.domain,
+          chainId: "1",
+        },
+      };
+
+      const signature = await signMessage(JSON.stringify(messageCopy.message));
+
+      // Verify the signature
+      await verifySignature(
+        signature,
+        account,
+        JSON.stringify(messageCopy.message),
+        parseInt(networkType || "1")
+      );
+    } catch (e: any) {
+      WatchTowerLogger.error(e.message, e);
+    }
+  }, [account, signMessage, verifySignature, networkType]);
+
+  let ethereumWalletDiv = (
+    <div>
+      {!account && (
+        <div>
+          <button
+            className={styles.connect}
+            onClick={connectWallet}
+            disabled={verifyingSignature}
+          >
+            {verifyingSignature ? "Connecting..." : "Connect Ethereum Wallet"}
+          </button>
+        </div>
+      )}
+      {account && !verifyingSignature && !verifiedSignature && (
+        <a className={styles.sign} onClick={sign}>
+          sign a message to verify your identity
+        </a>
+      )}
+      {verifyingSignature && (
+        <span className={styles.sign}>verifying your signature...</span>
+      )}
+      {unverifiedSignature && (
+        <div className="danger">
+          {getSignatureErrorMessage(unverifiedSignature).short}
+          <br />
+          {getSignatureErrorMessage(unverifiedSignature).advanced && (
+            <span className={styles.advancedErrorMessage}>
+              advanced: {getSignatureErrorMessage(unverifiedSignature).advanced}
+            </span>
+          )}
+        </div>
+      )}
+    </div>
+  );
+
+  return (
+    <div className={styles.verify}>
+      <Logo />
+      <div>
+        <DiscordServerInfo
```

Comment:
Can you apply all these components to the starknet page too ? The aim is to have reusable components, to avoid duplicated code